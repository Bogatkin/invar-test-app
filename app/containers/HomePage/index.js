/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Board from 'components/board';

class HomePage extends React.Component {

  static propTypes = {
    boards: React.PropTypes.object.isRequired,
    params: React.PropTypes.object.isRequired,
  };

  getCurrentBoard() {
    return this.props.boards.find((item) => (
      item.id === parseInt(this.props.params.boardId, 10)
    ));
  }

  render() {
    const board = this.getCurrentBoard();

    return (
      <div className="b-page">
        <h1 className="title">{board.name}</h1>
        <Board board={board} />
      </div>
    );
  }
}

export default connect(
  (state) => ({
    boards: state.get('app').get('boards'),
  }),
  (dispatch) => bindActionCreators({
    //
  }, dispatch)

)(HomePage);
