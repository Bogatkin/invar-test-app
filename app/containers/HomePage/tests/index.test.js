import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import { Map, List } from 'immutable';
import Board from 'models/board';

import HomePage from '../index';

const storeData = new Map({
  app: new Map({
    boards: new List([
      new Board({ id: 1, name: 'Test board 1' }),
      new Board({ id: 2, name: 'Test board 2' }),
    ]),
  }),
});
const mockStore = configureMockStore();

let store;

describe('<Nav />', () => {
  beforeEach(() => {
    store = mockStore(storeData);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<HomePage
      store={store}
      params={{ boardId: 1 }}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
