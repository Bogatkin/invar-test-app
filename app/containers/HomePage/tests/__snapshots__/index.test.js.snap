exports[`<Nav /> renders correctly 1`] = `
<div
  className="b-page">
  <h1
    className="title">
    Test board 1
  </h1>
  <div
    className="box b-board" />
</div>
`;
